### PR TITLE
added multi-gpu training support

### PR DESCRIPTION
**Pull request by toby-clark4**
Added multi-GPU support to the training script (on a single node), see issue #9. Currently requires truncation of the final batch with drop_last if the number of samples isn't divisible by global batch size, but can fix this with padding or similar in a future PR. I've tested the code with 4 GPUs for training and training and logging works as intended.

### CODE REVIEW

**Comment by Al-Murphy** [comment]
Hey @toby-clark4, we found two serious bugs in the implementation (see release notes if interested) but would you mind updating to thew latest version and submitting your changed to that? Cheers!

**Comment by toby-clark4** [comment]
Hey @Al-Murphy I think it should be updated now, sorry for the delay!

**Review by Al-Murphy — CHANGES_REQUESTED** [review]
Can you add documentation to the top of the train script explaining this is a DDP parallelisation approach (for example see here: https://lightning.ai/docs/pytorch/1.6.2/advanced/model_parallel.html) and how to run multi- vs single-gpu runs.

Secondly, did you test for parity of results when training on a single gpu vs multiple with this script?

**Comment by toby-clark4** [comment]
I'll add the documentation to the training script today. I tested a single epoch training run with global batch size of 2 on 1 vs 2 GPUs, which matched the loss on most but not all training steps (potentially because of per-device batchnorms?) and had the same validation loss values after training - is this sufficient or should I look into it further?

**Comment by Al-Murphy** [comment]
That's sufficient thanks for the work on this!!

**Comment by toby-clark4** [comment]
No problem! I've added explanations in the notes section of the train docstring, let me know if this is good or if I should move it somewhere else/ add some more detail.

**Review by Al-Murphy — APPROVED** [review]
Looks good, thanks for the work on this!
